src/avram/base_query_template.cr

```diff
@@ -48,20 +47,11 @@ class Avram::BaseQueryTemplate
           {{ column[:name] }}.eq(value)
         end
 
-        {% if column[:type].is_a?(Generic) %}
-          # Checking Array type
-          generate_criteria_method(BaseQuery, {{ column[:name] }}, {{ column[:type].type_vars.first }})
+        generate_criteria_method({{ column[:name] }}, {{ column[:type] }})
```

Comment:
I'm a little confused on why we don't need to check for `Array(String)` and it will know to use `String.adapter` 🤔

Reply:
Given a type of `Array(String)` it will call `Array(String).adapter` and that method calls `.adapter` on the generic type of the array which in this case is `String.adapter`.

https://github.com/luckyframework/avram/blob/efe1d8afaf337809c2accf51300daac48dba1cdc/src/avram/charms/array_extensions.cr#L1-L7

Reply:
ahh! boom! I missed that part.